script/tool/lib/src/build_examples_command.dart

```diff
@@ -49,164 +47,125 @@ class BuildExamplesCommand extends PluginCommand {
-      print(
+      printError(
           'None of ${platformSwitches.map((String platform) => '--$platform').join(', ')} '
-          'were specified, so not building anything.');
-      return;
+          'were specified. At least one platform must be provided.');
+      throw ToolExit(_exitNoPlatformFlags);
     }
-    final String flutterCommand =
-        const LocalPlatform().isWindows ? 'flutter.bat' : 'flutter';
-
-    final String enableExperiment = getStringArg(kEnableExperiment);
+  }
 
-    final List<String> failingPackages = <String>[];
-    await for (final Directory plugin in getPlugins()) {
-      for (final Directory example in getExamplesForPlugin(plugin)) {
-        final String packageName =
-            p.relative(example.path, from: packagesDir.path);
-
-        if (getBoolArg(kPlatformLinux)) {
-          print('\nBUILDING Linux for $packageName');
-          if (isLinuxPlugin(plugin)) {
-            final int buildExitCode = await processRunner.runAndStream(
-                flutterCommand,
-                <String>[
-                  'build',
-                  kPlatformLinux,
-                  if (enableExperiment.isNotEmpty)
-                    '--enable-experiment=$enableExperiment',
-                ],
-                workingDir: example);
-            if (buildExitCode != 0) {
-              failingPackages.add('$packageName (linux)');
-            }
-          } else {
-            print('Linux is not supported by this plugin');
+  @override
+  Future<List<String>> runForPackage(Directory package) async {
+    final List<String> errors = <String>[];
+
+    for (final Directory example in getExamplesForPlugin(package)) {
+      final String packageName =
+          p.relative(example.path, from: packagesDir.path);
+
+      if (getBoolArg(kPlatformLinux)) {
+        print('\nBUILDING $packageName for Linux');
+        if (isLinuxPlugin(package)) {
+          if (!await _buildExample(example, kPlatformLinux)) {
+            errors.add('$packageName (Linux)');
           }
+        } else {
+          printSkip('Linux is not supported by this plugin');
         }
+      }
 
-        if (getBoolArg(kPlatformMacos)) {
-          print('\nBUILDING macOS for $packageName');
-          if (isMacOsPlugin(plugin)) {
-            final int exitCode = await processRunner.runAndStream(
-                flutterCommand,
-                <String>[
-                  'build',
-                  kPlatformMacos,
-                  if (enableExperiment.isNotEmpty)
-                    '--enable-experiment=$enableExperiment',
-                ],
-                workingDir: example);
-            if (exitCode != 0) {
-              failingPackages.add('$packageName (macos)');
-            }
-          } else {
-            print('macOS is not supported by this plugin');
+      if (getBoolArg(kPlatformMacos)) {
+        print('\nBUILDING $packageName for macOS');
+        if (isMacOsPlugin(package)) {
```

Comment:
Should there be an else branch for isMacOsPlugin?  Printing out a warning?

Reply:
There is an `else`; it's the `printSkip`. It's not a warning because it's normal for many plugins not to support a given platform, particularly with federation.